conans/model/new_build_info.py

```diff
@@ -0,0 +1,172 @@
+import copy
+import os
+from collections import OrderedDict
+
+from conans.errors import ConanException
+from conans.model.build_info import DefaultOrderedDict
+
+_DIRS_VAR_NAMES = ["includedirs", "srcdirs", "libdirs", "resdirs", "bindirs", "builddirs",
+                  "frameworkdirs"]
+_FIELD_VAR_NAMES = ["system_libs", "frameworks", "libs", "defines", "cflags", "cxxflags",
+                   "sharedlinkflags", "exelinkflags"]
+
+
+class _BaseNewCppInfo(object):
+
+    def __init__(self):
+        self.names = {}
+        self.build_modules = {}
+
+        # ###### DIRECTORIES
+        self.includedirs = []  # Ordered list of include paths
+        self.srcdirs = []  # Ordered list of source paths
+        self.libdirs = []  # Directories to find libraries
+        self.resdirs = []  # Directories to find resources, data, etc
+        self.bindirs = []  # Directories to find executables and shared libs
+        self.builddirs = []
+        self.frameworkdirs = []
+
+        # ##### FIELDS
+        self.system_libs = []  # Ordered list of system libraries
+        self.frameworks = []  # Macos .framework
+        self.libs = []  # The libs to link against
+        self.defines = []  # preprocessor definitions
+        self.cflags = []  # pure C flags
+        self.cxxflags = []  # C++ compilation flags
+        self.sharedlinkflags = []  # linker flags
+        self.exelinkflags = []  # linker flags
+
+        self.sysroot = ""
+        self.requires = []
+
+        # ##### PATTERNS
+        self.include_patterns = []
+        self.lib_patterns = []
+        self.bin_patterns = []
+        self.src_patterns = []
+        self.build_patterns = []
+        self.res_patterns = []
+        self.framework_patterns = []
+
+    def get_name(self, generator):
+        return self.names.get(generator)
+
+    def to_absolute_paths(self, root_folder):
+        for name in _DIRS_VAR_NAMES:
+            setattr(self, name, [os.path.join(root_folder, c) for c in getattr(self, name)])
```

Comment:
Is it fully necessary that paths are converted to absolute paths? I can totally see a generator like CMakeDeps generating:

```cmake
set(MYPKG_ROOT_FOLDER /some/path/to/package/13234)
set(MYPKG_INCLUDEDIRS ${MYPKG_ROOT_FOLDER}/include ${MYPKG_ROOT_FOLDER}/myinc)
set(MYPKG_LIBDIRS ${MYPKG_ROOT_FOLDER)/lib)
```

Reply:
Totally right

Reply:
Done, review